### PR TITLE
Fix python3 incompatiable code

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
This is part of the python3 migration. 

I have fixed some compatibility warnings for python2 code. 

- print to function
- xrange -> range
- dict.iterXXX -> XXX

### CODE REVIEW

File: default/python/common/charting/charts.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 # Change dataDir to '<user>/AppData/Roaming/'
 
 
+from __future__ import print_function
```
Comment (Cjkjvfnby) [inline default/python/common/charting/charts.py:20]: Does anyone use this code?  It was initially written by @Dilvish-fo.

**Review by o01eg — APPROVED** [review]
I've built it and launched test game with 4 AI, on 69 turn AI take over my capital and test ends.

File: default/python/common/charting/charts.py (hunk 1 above)
Comment (o01eg) [reply default/python/common/charting/charts.py:20]: It's a script draws graphs based on logs (it don't work with non-ascii). I've tried it and it works for me with en_US locale.

File: default/python/turn_events/turn_events.py [hunk 2]
```diff
@@ -33,9 +33,9 @@ def execute_turn_events():
             x = radius + (dist_from_center * sin(angle))
             y = radius + (dist_from_center * cos(angle))
 
-        print "...creating new", field_type, "field, at distance", dist_from_center, "from center"
+        print("...creating new", field_type, "field, at distance", dist_from_center, "from center")
         if fo.create_field(field_type, x, y, size) == fo.invalid_object():
-            print >> sys.stderr, "Turn events: couldn't create new field"
+            print("Turn events: couldn't create new field")
```
Comment (Morlic-fo) [inline default/python/turn_events/turn_events.py]: Should probably keep logging errors to stderr here and the other places... ```print(..., file=sys.stderr)```

**Comment by Cjkjvfnby** [comment]
> I've built it and launched test game with 4 AI, on 69 turn AI take over my capital and test ends.

I don't know how to interpret this fact.  Is it good or bad?  

By the way, you don't need to *build* a game to test AI changes. 

I have my `persistent_config.xml` that points to the repository folder, so I just checkout branch and start installed game.

 ```
<?xml version="1.0"?>
<XMLDoc>
  <resource>
    <path>J:/projects/freeorion/default</path>
    <stringtable>
      <path>J:/projects/freeorion/default/stringtables/en.txt</path>
    </stringtable>
  </resource>
  <save>
    <path>j:/games/FreeOrion/save</path>
    <server>
      <path>j:/games/FreeOrion/save</path>
    </server>
  </save>
  </XMLDoc>
```

**Comment by o01eg** [comment]
> Is it good or bad?

It's good. AI can order and wasn't broken by changes.

File: default/python/universe_generation/starsystems.py [hunk 3]
```diff
@@ -115,7 +116,7 @@ def generate_systems(pos_list, gsd):
                 break
         else:
             # Intentionally non-modal.  Should be a warning.
-            print >> sys.stderr, ("Python generate_systems: place planets in system %d at position (%.2f, %.2f) failed"
-                                  % (system, position[0], position[1]))
+            print(("Python generate_systems: place planets in system %d at position (%.2f, %.2f) failed"
+                   % (system, position[0], position[1])))
```
Comment (Morlic-fo) [inline default/python/universe_generation/starsystems.py]: ^

File: default/python/universe_generation/util.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def report_error(msg):
     Handles error messages.
     """
     error_list.append(msg)
-    print >> sys.stderr, msg
+    print(msg)
```
Comment (Morlic-fo) [inline default/python/universe_generation/util.py]: ^

**Review by Morlic-fo — CHANGES_REQUESTED** [review]
Missed some logging to stderr.

**Review by Morlic-fo — APPROVED** [review]

**Comment by Morlic-fo** [comment]
Looks good to me.

**Pull request merged**